packages/app/pages/course/[cid]/invite.tsx

```diff
@@ -1,77 +1,156 @@
+
+  // If the user is already in the course, redirect to courses page
+  if (
+    profile &&
+    profile.courses.some((userCourse) => userCourse.course.id === Number(cid))
+  ) {
+    localStorage.removeItem("lastVisited");
+    router.push("/courses");
+  }
+
+  const renderCard = (
+    title,
+    description,
+    buttonLabel,
+    buttonAction,
+    cover = null
+  ) => (
+    <Card style={{ width: 600, textAlign: "center" }} cover={cover}>
+      <h1>{title}</h1>
+      <Space direction="vertical" style={{ width: "100%" }} size="large">
+        <Meta title={description.title} description={description.text} />
+        <Button
+          type="primary"
+          style={{ width: "100%", height: 50, borderRadius: "5px" }}
+          onClick={buttonAction}
+        >
+          {buttonLabel}
+        </Button>
+      </Space>
+    </Card>
+  );
+
+  if (isLoading) {
     return (
-      <>
-        <StandardPageContainer>
-          <Head>
-            <title>Page Not Found or is it?</title>
-          </Head>
-          page not found?
-        </StandardPageContainer>
-      </>
+      <div
+        style={{
+          display: "flex",
+          justifyContent: "center",
+          alignItems: "center",
+          height: "100vh",
+        }}
+      >
+        <Spin size="large" />
+      </div>
     );
-  } else {
-    return <Spin tip="Loading..." size="large" />;
   }
+
+  return (
+    <StandardPageContainer>
+      <Head>
+        <title>
+          {course ? `Invitation to join ‘${course?.name}‘` : "Invalid Invite"}
+        </title>
+      </Head>
+
+      <div
+        style={{
+          display: "flex",
+          justifyContent: "center",
+          alignItems: "center",
+          marginTop: 200,
+        }}
+      >
+        {profile?.organizationId !== course?.organizationCourse.id
+          ? renderCard(
+              "You cannot join a course that is not in your organization",
+              {},
+              "Back to my courses",
+              () => {
+                localStorage.removeItem("lastVisited");
+                router.push("/courses");
+              }
+            )
+          : code !== course?.courseInviteCode
+          ? renderCard("Invalid Course Code", {}, "Back to my courses", () => {
+              localStorage.removeItem("lastVisited");
+              router.push("/courses");
+            })
+          : renderCard(
+              `Invitation to join ‘${course?.name}‘`,
+              { title: cardMetaTitle, text: cardMetaDescription },
+              "Accept Invitation",
+              async () => {
+                const userData: UBCOuserParam = {
+                  email: profile.email,
+                  first_name: profile.firstName ?? "",
+                  password: "",
+                  last_name: profile.lastName ?? "",
+                  selected_course: course.id,
+                  sid: profile.sid,
+                  photo_url: profile.photoURL,
+                  courses: [],
```

Comment:
What if user is registered in the course?

Reply:
They will be redirected to courses page.